lesson36/src/js/article.js

```diff
@@ -0,0 +1,114 @@
+  element.className = className;
+  return element;
+};
+
+const createErrorMessage = (text) => {
+  const errorMessage = createElementWithClassName("p", "article-error-message");
+  errorMessage.textContent = text;
+  return errorMessage;
+};
+
+const renderLoading = () => {
+  const loadingWrapper = createElementWithClassName("div", "article-loading-wrapper");
+  const loading = document.createElement("img");
+  loadingWrapper.id = "js-loading";
+  loading.src = "./img/article-loading.gif";
+  articleList.appendChild(loadingWrapper).appendChild(loading);
+};
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (endpoint) => {
+  const response = await fetch(endpoint);
+  if (!response.ok) {
+    const responseMessage = `${response.status}:${response.statusText}`;
+    articleList.append(createErrorMessage(responseMessage));
+    console.error(responseMessage);
+    return;
+  }
+  return await response.json();
+}
+
+const getArticleData = async (api) => {
+  try {
+    return await fetchData(api);
+  } catch (error) {
+    console.error(error);
+    articleList.before(createErrorMessage(error));
+  }
+}
+
+const renderArticleItems = (articleData) => {
+  articleData.forEach((data) => {
+    const item = createElementWithClassName("li", "article__item");
+    const content = createElementWithClassName("p", "article__item-content");
+    const author = document.createElement("p");
+    author.textContent = data.author;
+    content.textContent = data.quote;
+    articleList.appendChild(item).appendChild(author).after(content);
+  });
+}
+
+let totalPosts;
+const endpointConfig = {
+  path: "https://api.javascripttutorial.net/v1/quotes/",
+  limit: 10,
+  currentPage: 1,
+  setEndpoint: () => {
+    const url = new URL(endpointConfig.path);
+    url.searchParams.set("page", endpointConfig.currentPage);
+    url.searchParams.set("limit", endpointConfig.limit);
+    return url.href;
+  }
+}
+
+const observeConfig = {
+  observe: null,
+  option: { threshold: 1.0 },
+  startObserve: (target, callback) => {
+    observeConfig.observe = new IntersectionObserver(callback, observeConfig.option);
+    observeConfig.observe.observe(target);
+  },
+  stopObserve: (target) => {
+    observeConfig.observe.unobserve(target);
+  }
+}
+
+const getArticleDataAndUpdate = async () => {
+  ++endpointConfig.currentPage;
+  const articleData = await getArticleData(endpointConfig.setEndpoint());
+  removeLoading();
+  if (articleData) {
+    renderArticleItems(articleData.data);
+    observeConfig.startObserve(articleList.lastElementChild, intersectHandler);
+  }
+}
+
+const intersectHandler = ([entry]) => {
+  if (!entry.isIntersecting) return;
+  observeConfig.stopObserve(entry.target);
+  const articleItems = document.querySelectorAll(".article__item");
+  if (articleItems.length < totalPosts) {
+    renderLoading();
+    setTimeout(getArticleDataAndUpdate, 500);
+  }
+}
```

Comment:
Q. I could not understand the function here.. What is [entry] coming in as an argument? 
Sorry for the lack of study.

Reply:
[IntersectionObserverEntry](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry) is passed as a callback argument.It is easier to understand the properties included if you look at the MDN.

The IntersectionObserverEntry object stores the number of elements to be monitored in an array.
If there are three objects to be monitored, the situation is as follows.
```js
 [IntersectionObserverEntry, IntersectionObserverEntry, IntersectionObserverEntry]
```
In this implementation, since there is only one target to monitor, a split allocation was performed and implemented
Are your questions adequately answered...?

Reply:
ご説明ありがとうございます！！
なるほどです、勉強になります:eyes: Intersection Observerに初めて触れたのですがご説明いただき理解が進みました！！自分でももっと調べてみようと思いました、ありがとうございます！